ui/apps/platform/src/hooks/useFetchNamespaceDeployments.ts

```diff
@@ -1,68 +1,88 @@
 
-import queryService from 'utils/queryService';
+import { listDeployments } from 'services/DeploymentsService';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
 
 type Deployment = {
     id: string;
     name: string;
 };
 export type NamespaceWithDeployments = {
     metadata: {
-        id: string;
         name: string;
     };
     deployments: Deployment[];
 };
-type DeploymentResponse = {
-    results: NamespaceWithDeployments[];
-};
 
-const DEPLOYMENTS_FOR_NAMESPACE_QUERY = gql`
-    query getNamespaceDeployments($query: String!) {
-        results: namespaces(query: $query) {
-            metadata {
-                name
-                id
-            }
-            deployments {
-                name
-                id
-            }
-        }
-    }
-`;
+type ListDeploymentResponse = {
+    loading: boolean;
+    error: string;
+    deploymentsByNamespace: NamespaceWithDeployments[];
+};
 
 function useFetchNamespaceDeployments(selectedNamespaceIds: string[]) {
-    const [availableDeployments, setAvailableNamespaces] = useState<NamespaceWithDeployments[]>([]);
-
-    const searchClause = { 'Namespace ID': selectedNamespaceIds };
-    // If the selectedNamespaceId has not been set yet, do not run the gql query
-    const queryOptions =
-        selectedNamespaceIds.length > 0
-            ? { variables: { query: queryService.objectToWhereClause(searchClause) } }
-            : { skip: true };
-
-    const { loading, error, data } = useQuery<DeploymentResponse, { query: string }>(
-        DEPLOYMENTS_FOR_NAMESPACE_QUERY,
-        queryOptions
-    );
-
+    const [deploymentResponse, setDeploymentResponse] = useState<ListDeploymentResponse>({
+        loading: false,
+        error: '',
+        deploymentsByNamespace: [],
+    });
+    const selectedNamespaceIdsAsString = JSON.stringify(selectedNamespaceIds);
     useEffect(() => {
-        if (!data || !data.results) {
-            setAvailableNamespaces([]);
-        } else {
-            setAvailableNamespaces(data.results);
-        }
-        // clean up state on unmount
-        return () => setAvailableNamespaces([]);
-    }, [data]);
+        const namespaceIds = JSON.parse(selectedNamespaceIdsAsString);
+        if (namespaceIds.length > 0) {
+            setDeploymentResponse({
+                loading: true,
+                error: '',
+                deploymentsByNamespace: [],
+            });
+            const searchQuery: Record<string, string[]> = {
+                'Namespace ID': namespaceIds,
+            };
+            const sortOption = { field: 'Deployment', reversed: 'false' };
+            listDeployments(searchQuery, sortOption, 0, 0)
+                .then((response) => {
+                    const namespacesWithDeployments: NamespaceWithDeployments[] = [];
+                    const deploymentsByNamespace = new Map<string, Deployment[]>();
+                    response.forEach(({ id, name, namespace }) => {
```

Comment:
The underscore code would look like this.
```
                    const deploymentsIndexesByNamespace = _.groupBy(response, 'namespace');
                    const namespaces = _.keys(deploymentsIndexesByNamespace);
                    _.each(namespaces, (ns) => {
                        const namespaceDeployments: Deployment[] = [];
                        _.each(deploymentsIndexesByNamespace[ns], ({ id, name }) => {
                            namespaceDeployments.push({ id, name });
                        });
                        const namespaceWithDeployments: NamespaceWithDeployments = {
                            metadata: {
                                name: ns,
                            },
                            deployments: namespaceDeployments,
                        };
                        namespacesWithDeployments.push(namespaceWithDeployments);
                    });
```
I get the following error with the above code:
`TS2686: '_' refers to a UMD global, but the current file is a module. Consider adding an import instead`

Reply:
Instead of underscore, UI has lodash as a dependency. If you decide to use its `groupBy` method:

https://lodash.com/docs/4.17.15#groupBy

then here is the import statement:

```js
import groupBy from 'lodash/groupBy';
```

Reply:
Whether or not you use lodash `groupBy` method, what do you think about factoring out as a function (in this file, but outside of the hook) which returns `NamespaceWithDeployments[]` for `deployments: ListDeployment` argument.